--- a/cmake/modules/FindKDE4Internal.cmake
+++ b/cmake/modules/FindKDE4Internal.cmake
@@ -666,25 +666,6 @@ if(_kdeBootStrapping  OR  TARGET ${KDE4_
    _kde4_set_lib_variables(NEPOMUK nepomuk "${KDE4_TARGET_PREFIX}")
 endif(_kdeBootStrapping  OR  TARGET ${KDE4_TARGET_PREFIX}nepomuk)
 
-
-################### try to find Phonon ############################################
-
-# we do this here instead of above together with the checks for Perl etc.
-# since FindPhonon.cmake also uses ${KDE4_LIB_INSTALL_DIR} to check for Phonon,
-# which helps with finding the phonon installed as part of kdesupport:
-
-# only make Phonon REQUIRED if KDE4 itself is REQUIRED
-find_package(Phonon 4.3.80 ${_REQ_STRING_KDE4})
-set(KDE4_PHONON_LIBRARY ${PHONON_LIBRARY})
-set(KDE4_PHONON_LIBS ${PHONON_LIBS})
-set(KDE4_PHONON_INCLUDES ${PHONON_INCLUDES})
-
-if(NOT PHONON_FOUND)
-   message(STATUS "KDE4 not found, because Phonon was not found")
-   return()
-endif(NOT PHONON_FOUND)
-
-
 #####################  provide some options   ##########################################
 
 option(KDE4_ENABLE_FINAL "Enable final all-in-one compilation")
--- a/interfaces/CMakeLists.txt
+++ b/interfaces/CMakeLists.txt
@@ -4,12 +4,10 @@ project(interfaces)
 
 add_subdirectory( ktexteditor ) 
 add_subdirectory( kregexpeditor ) 
-add_subdirectory( kmediaplayer ) 
 add_subdirectory( kio ) 
 add_subdirectory( terminal ) 
 add_subdirectory( khexedit ) 
 add_subdirectory( kimproxy ) 
-add_subdirectory( kspeech ) 
 
 ########### install files ###############
 
--- a/knotify/config/knotifyconfigactionswidget.cpp
+++ b/knotify/config/knotifyconfigactionswidget.cpp
@@ -21,7 +21,10 @@
 #include <kstandarddirs.h>
 #include <kiconloader.h>
 
+// ALERT Phonon support has been removed by zaufi's patch
+#if 0
 #include <phonon/mediaobject.h>
+#endif
 
 KNotifyConfigActionsWidget::KNotifyConfigActionsWidget( QWidget * parent )
 	: QWidget(parent)
@@ -125,6 +128,8 @@ void KNotifyConfigActionsWidget::save( K
 
 void KNotifyConfigActionsWidget::slotPlay(  )
 {
+// ALERT Phonon support has been removed by zaufi's patch
+#if 0
 	KUrl soundURL = m_ui.Sound_select->url();
 	if ( soundURL.isRelative() )
 	{
@@ -138,6 +143,7 @@ void KNotifyConfigActionsWidget::slotPla
 	Phonon::MediaObject* media = Phonon::createPlayer( Phonon::NotificationCategory, soundURL );
 	media->play();
 	connect(media, SIGNAL(finished()), media, SLOT(deleteLater()));
+#endif
 }
 
 void KNotifyConfigActionsWidget::slotKTTSComboChanged()
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,9 +250,9 @@ set(KDE4_KDE3SUPPORT_INCLUDES ${CMAKE_SO
                               ${CMAKE_SOURCE_DIR}/kde3support/kio
                               ${KDE4_KPARTS_INCLUDES})
 
-if(NOT WINCE)
- set(KDE4_KHTML_INCLUDES ${CMAKE_SOURCE_DIR}/khtml)
-endif(NOT WINCE)
+#if(NOT WINCE)
+# set(KDE4_KHTML_INCLUDES ${CMAKE_SOURCE_DIR}/khtml)
+#endif(NOT WINCE)
 
 
 ################# configure checks and create the configured files #################
@@ -338,9 +338,9 @@ add_subdirectory( mimetypes  )
 add_subdirectory( kinit  )
 add_subdirectory( threadweaver )
 add_subdirectory( sonnet  )
-if(NOT WINCE)
-add_subdirectory( khtml  )
-endif(NOT WINCE)
+#if(NOT WINCE)
+#add_subdirectory( khtml  )
+#endif(NOT WINCE)
 add_subdirectory( interfaces  )
 #if ( NOT CMAKE_CROSSCOMPILING AND QT_QTDESIGNER_FOUND )
 add_subdirectory( kdewidgets  )
