--- configure.ac.org	2013-01-28 03:36:42.000000000 +0400
+++ configure.ac	2013-07-30 08:27:04.000000000 +0400
@@ -291,8 +291,7 @@
 fi
 HAVE_BTRFS="yes"
 AC_PATH_PROG([BTRFS], [btrfs], [], [$PATH:/sbin:/usr/sbin])
-AC_PATH_PROG([BTRFSCTL], [btrfsctl], [], [$PATH:/sbin:/usr/sbin])
-if test -z "$BTRFS" || test -z "$BTRFSCTL"; then
+if test -z "$BTRFS"; then
   HAVE_BTRFS="no"
 fi
 HAVE_LOOPBACK="yes"
