--- a/cmake/modules/FindGStreamer.cmake
+++ b/cmake/modules/FindGStreamer.cmake
@@ -15,6 +15,7 @@
 #
 # Copyright (c) 2010, Collabora Ltd.
 #   @author George Kiagiadakis <george.kiagiadakis@collabora.co.uk>
+# Copyright (c) 2016, Alex Turbov <i.zaufi@gmail.com>
 #
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
@@ -38,14 +39,22 @@ if (PKG_CONFIG_FOUND)
                  OUTPUT_VARIABLE PKG_GSTREAMER_PLUGIN_DIR)
 endif()
 
-find_library(GSTREAMER_LIBRARY
-             NAMES gstreamer-${GSTREAMER_ABI_VERSION}
-             HINTS ${PKG_GSTREAMER_LIBRARY_DIRS} ${PKG_GSTREAMER_LIBDIR})
-
-find_path(GSTREAMER_INCLUDE_DIR
-          gst/gst.h
-          HINTS ${PKG_GSTREAMER_INCLUDE_DIRS} ${PKG_GSTREAMER_INCLUDEDIR}
-          PATH_SUFFIXES gstreamer-${GSTREAMER_ABI_VERSION})
+# Use `pkg-config` to obtain flags if found
+if(PKG_CONFIG_FOUND)
+    # Just use obtained variables
+    set(GSTREAMER_LIBRARY ${PKG_GSTREAMER_LIBRARIES})
+    set(GSTREAMER_INCLUDE_DIR ${PKG_GSTREAMER_INCLUDE_DIRS})
+else()
+    # Use old way if not found
+    find_library(GSTREAMER_LIBRARY
+                NAMES gstreamer-${GSTREAMER_ABI_VERSION}
+                HINTS ${PKG_GSTREAMER_LIBRARY_DIRS} ${PKG_GSTREAMER_LIBDIR})
+
+    find_path(GSTREAMER_INCLUDE_DIR
+            gst/gst.h
+            HINTS ${PKG_GSTREAMER_INCLUDE_DIRS} ${PKG_GSTREAMER_INCLUDEDIR}
+            PATH_SUFFIXES gstreamer-${GSTREAMER_ABI_VERSION})
+endif()
 
 if (PKG_GSTREAMER_PLUGIN_DIR)
     set(_GSTREAMER_PLUGIN_DIR ${PKG_GSTREAMER_PLUGIN_DIR})
