diff -u -rub paludis-2.4.0.org/paludis/resolver/collect_depped_upon.cc paludis-2.4.0/paludis/resolver/collect_depped_upon.cc
--- paludis-2.4.0.org/paludis/resolver/collect_depped_upon.cc	2015-05-03 13:07:27.000000000 +0800
+++ paludis-2.4.0/paludis/resolver/collect_depped_upon.cc	2013-12-30 03:40:26.000000000 +0800
@@ -239,7 +239,7 @@
     }
     else
     {
-        auto k { &PackageID::build_dependencies_key, &PackageID::run_dependencies_key, &PackageID::post_dependencies_key };
+        auto k = { &PackageID::build_dependencies_key, &PackageID::run_dependencies_key, &PackageID::post_dependencies_key };
         for (auto i(k.begin()), i_end(k.end()) ; i != i_end ; ++i)
         {
             auto key(((*id).*(*i))());
