diff -u -rub paludis-2.4.0.org/paludis/syncers/dorsync paludis-2.4.0/paludis/syncers/dorsync
--- paludis-2.4.0.org/paludis/syncers/dorsync	2015-04-02 03:09:36.000000000 +0800
+++ paludis-2.4.0/paludis/syncers/dorsync	2015-04-13 16:18:10.000000000 +0800
@@ -89,7 +89,7 @@
 REMOTE="${REMOTE#rsync+ssh://}"
 
 ${RSYNC_WRAPPER} rsync --recursive --links --safe-links --perms --times \
-    --new-compress --force --whole-file --delete --delete-delay --stats --timeout=180 \
+    --compress --force --whole-file --delete --delete-delay --stats --timeout=180 \
     ${PALUDIS_SYNC_FILTER_FILE:+--filter "merge ${PALUDIS_SYNC_FILTER_FILE}"} \
     --exclude=/.cache --progress "${RSYNC_OPTIONS[@]}" "${REMOTE%/}/" "${LOCAL}/"
 
