--- a/setup.py
+++ b/setup.py
@@ -39,8 +39,7 @@ setup(name='urllib3',
       url='https://urllib3.readthedocs.io/',
       license='MIT',
       packages=['urllib3',
-                'urllib3.packages', 'urllib3.packages.ssl_match_hostname',
-                'urllib3.packages.backports', 'urllib3.contrib',
+                'urllib3.contrib',
                 'urllib3.util',
                 ],
       requires=[],
