--- a/setup.py
+++ b/setup.py
@@ -39,8 +39,8 @@ setup(name='urllib3',
       url='https://urllib3.readthedocs.io/',
       license='MIT',
       packages=['urllib3',
+                'urllib3.contrib',
                 'urllib3.packages', 'urllib3.packages.ssl_match_hostname',
-                'urllib3.packages.backports', 'urllib3.contrib',
                 'urllib3.util',
                 ],
       requires=[],
--- a/urllib3/_collections.py
+++ b/urllib3/_collections.py
@@ -11,11 +11,8 @@ except ImportError:  # Platform-specific
             pass
 
 
-try:  # Python 2.7+
-    from collections import OrderedDict
-except ImportError:
-    from .packages.ordered_dict import OrderedDict
-from .packages.six import iterkeys, itervalues, PY3
+from collections import OrderedDict
+from six import iterkeys, itervalues, PY3
 
 
 __all__ = ['RecentlyUsedContainer', 'HTTPHeaderDict']
--- a/urllib3/connectionpool.py
+++ b/urllib3/connectionpool.py
@@ -1,6 +1,7 @@
 from __future__ import absolute_import
 import errno
 import logging
+import six
 import sys
 import warnings
 
@@ -31,7 +32,6 @@ from .exceptions import (
     NewConnectionError,
 )
 from .packages.ssl_match_hostname import CertificateError
-from .packages import six
 from .connection import (
     port_by_scheme,
     DummyConnection,
--- a/urllib3/connection.py
+++ b/urllib3/connection.py
@@ -2,11 +2,11 @@ from __future__ import absolute_import
 import datetime
 import logging
 import os
+import six
 import sys
 import socket
 from socket import error as SocketError, timeout as SocketTimeout
 import warnings
-from .packages import six
 
 try:  # Python 3
     from http.client import HTTPConnection as _HTTPConnection
--- a/urllib3/fields.py
+++ b/urllib3/fields.py
@@ -1,8 +1,7 @@
 from __future__ import absolute_import
 import email.utils
 import mimetypes
-
-from .packages import six
+import six
 
 
 def guess_content_type(filename, default='application/octet-stream'):
--- a/urllib3/filepost.py
+++ b/urllib3/filepost.py
@@ -4,8 +4,8 @@ import codecs
 from uuid import uuid4
 from io import BytesIO
 
-from .packages import six
-from .packages.six import b
+import six
+from six import b
 from .fields import RequestField
 
 writer = codecs.lookup('utf-8')[3]
--- a/urllib3/response.py
+++ b/urllib3/response.py
@@ -9,8 +9,8 @@ from ._collections import HTTPHeaderDict
 from .exceptions import (
     ProtocolError, DecodeError, ReadTimeoutError, ResponseNotChunked
 )
-from .packages.six import string_types as basestring, binary_type, PY3
-from .packages.six.moves import http_client as httplib
+from six import string_types as basestring, binary_type, PY3
+from six.moves import http_client as httplib
 from .connection import HTTPException, BaseSSLError
 from .util.response import is_fp_closed, is_response_to_head
 
