diff --git a/Makefile b/Makefile
index c43dcfd..c9b28b6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,7 @@
 obj-m := system76.o
-KVERSION := $(shell uname -r)

 all:
-       $(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+       $(MAKE) -C "$(KERNEL_DIR)" M="$(PWD)" modules

 clean:
-       $(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
+       $(MAKE) -C "$(KERNEL_DIR)" M="$(PWD)" clean
