--- a/ace/SOCK_Dgram_Mcast.cpp
+++ b/ace/SOCK_Dgram_Mcast.cpp
@@ -183,11 +183,13 @@ ACE_SOCK_Dgram_Mcast::open_i (const ACE_
     {
 #if defined (SO_REUSEPORT)
       int one = 1;
-      if (this->ACE_SOCK::set_option (SOL_SOCKET,
+      // NOTE The only possible error code here is ENOPROTOOPT
+      // (overridden by ACE to ENOTSUP) which means that option
+      // is not supported. And it is Ok to ignore this (bonus option).
+      this->ACE_SOCK::set_option (SOL_SOCKET,
                                       SO_REUSEPORT,
                                       &one,
-                                      sizeof one) == -1)
-        return -1;
+                                      sizeof one);
 #endif /* SO_REUSEPORT */
     }
 
