--- a/ace/SSL/SSL_Context.cpp
+++ b/ace/SSL/SSL_Context.cpp
@@ -244,15 +244,6 @@ ACE_SSL_Context::set_mode (int mode)
 
   switch (mode)
     {
-    case ACE_SSL_Context::SSLv2_client:
-      method = ::SSLv2_client_method ();
-      break;
-    case ACE_SSL_Context::SSLv2_server:
-      method = ::SSLv2_server_method ();
-      break;
-    case ACE_SSL_Context::SSLv2:
-      method = ::SSLv2_method ();
-      break;
     case ACE_SSL_Context::SSLv3_client:
       method = ::SSLv3_client_method ();
       break;
@@ -348,9 +339,7 @@ ACE_SSL_Context::load_trusted_ca (const
       || mode_ == TLSv1
       || mode_ == TLSv1_server
       || mode_ == SSLv3
-      || mode_ == SSLv3_server
-      || mode_ == SSLv2
-      || mode_ == SSLv2_server)
+      || mode_ == SSLv3_server)
     {
       // Note: The STACK_OF(X509_NAME) pointer is a copy of the pointer in
       // the CTX; any changes to it by way of these function calls will
--- a/ace/SSL/SSL_Context.h
+++ b/ace/SSL/SSL_Context.h
@@ -82,10 +82,7 @@ public:
 
   enum {
     INVALID_METHOD = -1,
-    SSLv2_client = 1,
-    SSLv2_server,
-    SSLv2,
-    SSLv3_client,
+    SSLv3_client = 4,
     SSLv3_server,
     SSLv3,
     SSLv23_client,
