diff -u -r webkitgtk-2.2.4.org/GNUmakefile.am webkitgtk-2.2.4/GNUmakefile.am
--- webkitgtk-2.2.4.org/GNUmakefile.am	2014-01-20 16:06:03.000000000 +0400
+++ webkitgtk-2.2.4/GNUmakefile.am	2014-02-03 23:47:21.000000000 +0400
@@ -277,3 +277,11 @@
 	@-cp ./Documentation/webkit2gtk/tmpl/*.sgml $(distdir)/Documentation/webkit2gtk/tmpl
 	@cp ./Documentation/webkit2gtk/html/* $(distdir)/Documentation/webkit2gtk/html
 endif
+
+all-built-sources-local: $(BUILT_SOURCES) autotoolsconfig.h
+
+all-ltlibraries-local: GNUmakefile $(LTLIBRARIES)
+
+all-programs-local: GNUmakefile $(PROGRAMS)
+
+all-data-local: GNUmakefile $(DATA)
diff -u -r webkitgtk-2.2.4.org/Source/autotools/SetupCompilerFlags.m4 webkitgtk-2.2.4/Source/autotools/SetupCompilerFlags.m4
--- webkitgtk-2.2.4.org/Source/autotools/SetupCompilerFlags.m4	2014-01-20 11:57:54.000000000 +0400
+++ webkitgtk-2.2.4/Source/autotools/SetupCompilerFlags.m4	2014-02-04 00:57:08.027671657 +0400
@@ -16,10 +16,10 @@
 fi
 
 if test "$c_compiler" = "gcc"; then
-    CFLAGS="$CFLAGS -D_FORTIFY_SOURCE=2"
+    CFLAGS="$CFLAGS -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2"
 fi
 if test "$cxx_compiler" = "g++"; then
-    CXXFLAGS="$CXXFLAGS -D_FORTIFY_SOURCE=2"
+    CXXFLAGS="$CXXFLAGS -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2"
 fi
 
 if test "$host_cpu" = "sh4"; then
diff -u -r webkitgtk-2.2.4.org/Source/WebCore/bindings/gobject/GNUmakefile.am webkitgtk-2.2.4/Source/WebCore/bindings/gobject/GNUmakefile.am
--- webkitgtk-2.2.4.org/Source/WebCore/bindings/gobject/GNUmakefile.am	2014-01-20 11:57:46.000000000 +0400
+++ webkitgtk-2.2.4/Source/WebCore/bindings/gobject/GNUmakefile.am	2014-02-04 00:31:07.967339812 +0400
@@ -1,4 +1,5 @@
 webkitgtk_gdom_built_sources += \
+	DerivedSources/webkitdom/$(am__dirstamp) \
 	DerivedSources/webkitdom/WebKitDOMAttr.cpp \
 	DerivedSources/webkitdom/WebKitDOMAttrPrivate.h \
 	DerivedSources/webkitdom/WebKitDOMBarProp.cpp \
@@ -427,6 +428,7 @@
 
 if ENABLE_VIDEO
 webkitgtk_gdom_built_h_api += \
+	$(top_builddir)/DerivedSources/webkitdom/$(am__dirstamp) \
 	$(top_builddir)/DerivedSources/webkitdom/WebKitDOMAudioTrack.h \
 	$(top_builddir)/DerivedSources/webkitdom/WebKitDOMAudioTrackList.h \
 	$(top_builddir)/DerivedSources/webkitdom/WebKitDOMHTMLAudioElement.h \
@@ -439,6 +441,7 @@
 	$(top_builddir)/DerivedSources/webkitdom/WebKitDOMVideoTrack.h \
 	$(top_builddir)/DerivedSources/webkitdom/WebKitDOMVideoTrackList.h
 webkitgtk_gdom_built_sources += \
+	DerivedSources/webkitdom/$(am__dirstamp) \
 	DerivedSources/webkitdom/WebKitDOMHTMLAudioElement.cpp \
 	DerivedSources/webkitdom/WebKitDOMHTMLAudioElementPrivate.h \
 	DerivedSources/webkitdom/WebKitDOMHTMLVideoElement.cpp \
diff -u -r webkitgtk-2.2.4.org/Source/WebCore/GNUmakefile.am webkitgtk-2.2.4/Source/WebCore/GNUmakefile.am
--- webkitgtk-2.2.4.org/Source/WebCore/GNUmakefile.am	2014-01-20 11:57:46.000000000 +0400
+++ webkitgtk-2.2.4/Source/WebCore/GNUmakefile.am	2014-02-04 03:37:25.038438357 +0400
@@ -358,6 +358,12 @@
 	$(AM_V_GEN)$(BISON) --no-lines --defines="DerivedSources/ANGLE/glslang_tab.h" --skeleton=yacc.c --output="$@" $<
 DerivedSources/ANGLE/glslang_tab.h: DerivedSources/ANGLE/glslang_tab.cpp
 
+BUILT_SOURCES += \
+	DerivedSources/ANGLE/$(am__dirstamp) \
+	DerivedSources/ANGLE/glslang.cpp \
+	DerivedSources/ANGLE/glslang_tab.cpp \
+	DerivedSources/ANGLE/glslang_tab.h
+
 IDL_PATH := \
     $(WebCore)/Modules/battery \
     $(WebCore)/Modules/filesystem \
diff -u -r webkitgtk-2.2.4.org/Source/WebCore/GNUmakefile.list.am webkitgtk-2.2.4/Source/WebCore/GNUmakefile.list.am
--- webkitgtk-2.2.4.org/Source/WebCore/GNUmakefile.list.am	2014-01-20 11:57:46.000000000 +0400
+++ webkitgtk-2.2.4/Source/WebCore/GNUmakefile.list.am	2014-02-04 00:26:45.946502487 +0400
@@ -1,4 +1,5 @@
 webcore_built_sources += \
+	DerivedSources/WebCore/$(am__dirstamp) \
 	DerivedSources/WebCore/CSSGrammar.cpp \
 	DerivedSources/WebCore/CSSGrammar.h \
 	DerivedSources/WebCore/CSSPropertyNames.h \
@@ -875,6 +876,7 @@
 	DerivedSources/WebCore/XPathGrammar.h
 
 platform_built_sources += \
+	DerivedSources/Platform/$(am__dirstamp) \
 	DerivedSources/Platform/ColorData.cpp \
 	DerivedSources/Platform/WebKitFontFamilyNames.cpp \
 	DerivedSources/Platform/WebKitFontFamilyNames.h
@@ -882,12 +884,14 @@
 # These files need to be part of WebCore otherwise they cause undefined
 # symbols havoc
 webcore_built_sources += \
+	DerivedSources/WebCore/$(am__dirstamp) \
 	DerivedSources/WebCore/JSSVGDocument.cpp \
 	DerivedSources/WebCore/JSSVGDocument.h \
 	DerivedSources/WebCore/SVGNames.cpp \
 	DerivedSources/WebCore/SVGElementFactory.cpp
 
 webcore_svg_built_sources += \
+	DerivedSources/WebCore/$(am__dirstamp) \
 	DerivedSources/WebCore/JSSVGAElement.cpp \
 	DerivedSources/WebCore/JSSVGAElement.h \
 	DerivedSources/WebCore/JSSVGAltGlyphDefElement.cpp \
diff -u -r webkitgtk-2.2.4.org/Source/WebInspectorUI/GNUmakefile.am webkitgtk-2.2.4/Source/WebInspectorUI/GNUmakefile.am
--- webkitgtk-2.2.4.org/Source/WebInspectorUI/GNUmakefile.am	2014-01-20 11:57:53.000000000 +0400
+++ webkitgtk-2.2.4/Source/WebInspectorUI/GNUmakefile.am	2014-02-04 00:14:53.820240238 +0400
@@ -27,9 +27,11 @@
 		--target=${GENSOURCES_WEBINSPECTOR_UI}/GResourceBundle.c ${GENSOURCES_WEBINSPECTOR_UI}/GResourceBundle.xml
 
 BUILT_SOURCES += \
+	DerivedSources/WebInspectorUI/$(am__dirstamp) \
 	DerivedSources/WebInspectorUI/GResourceBundle.xml
 
 webcore_built_sources += \
+	DerivedSources/WebInspectorUI/$(am__dirstamp) \
 	DerivedSources/WebInspectorUI/GResourceBundle.c
 
 EXTRA_DIST += \
diff -u -r webkitgtk-2.2.4.org/Source/WebKit/gtk/GNUmakefile.am webkitgtk-2.2.4/Source/WebKit/gtk/GNUmakefile.am
--- webkitgtk-2.2.4.org/Source/WebKit/gtk/GNUmakefile.am	2014-01-21 15:41:08.000000000 +0400
+++ webkitgtk-2.2.4/Source/WebKit/gtk/GNUmakefile.am	2014-02-04 00:13:19.217462448 +0400
@@ -176,6 +176,7 @@
 # webkitenumtypes.{h,cpp} cannot be in webkitgtk_built_h_api, because
 # the rule that builds it depends on webkitgtk_built_h_api (circular dependency).
 webkitgtk_built_sources += \
+	DerivedSources/webkit/$(am__dirstamp) \
 	DerivedSources/webkit/webkitenumtypes.cpp \
 	DerivedSources/webkit/webkitenumtypes.h \
 	DerivedSources/webkit/webkitmarshal.cpp \
diff -u -r webkitgtk-2.2.4.org/Tools/GNUmakefile.am webkitgtk-2.2.4/Tools/GNUmakefile.am
--- webkitgtk-2.2.4.org/Tools/GNUmakefile.am	2014-01-20 11:57:54.000000000 +0400
+++ webkitgtk-2.2.4/Tools/GNUmakefile.am	2014-02-04 00:07:53.194117947 +0400
@@ -64,6 +64,7 @@
 	Source/WebCore/testing/js/WebCoreTestSupport.h
 
 libwebcoreinternals_built_sources += \
+	DerivedSources/WebCore/$(am__dirstamp) \
 	DerivedSources/WebCore/InternalSettingsGenerated.cpp \
 	DerivedSources/WebCore/InternalSettingsGenerated.h   \
 	DerivedSources/WebCore/JSMallocStatistics.cpp \
diff -u -r webkitgtk-2.2.4.org/Tools/MiniBrowser/gtk/GNUmakefile.am webkitgtk-2.2.4/Tools/MiniBrowser/gtk/GNUmakefile.am
--- webkitgtk-2.2.4.org/Tools/MiniBrowser/gtk/GNUmakefile.am	2014-01-20 11:57:54.000000000 +0400
+++ webkitgtk-2.2.4/Tools/MiniBrowser/gtk/GNUmakefile.am	2014-02-04 00:08:54.265684212 +0400
@@ -30,6 +30,7 @@
 	Tools/MiniBrowser/gtk/main.c
 
 minibrowser_built_sources += \
+	DerivedSources/WebKit2/$(am__dirstamp) \
 	DerivedSources/WebKit2/BrowserMarshal.h \
 	DerivedSources/WebKit2/BrowserMarshal.c
 nodist_Programs_MiniBrowser_SOURCES = \
diff -u -r webkitgtk-2.2.4.org/Tools/WebKitTestRunner/GNUmakefile.am webkitgtk-2.2.4/Tools/WebKitTestRunner/GNUmakefile.am
--- webkitgtk-2.2.4.org/Tools/WebKitTestRunner/GNUmakefile.am	2014-01-20 11:57:54.000000000 +0400
+++ webkitgtk-2.2.4/Tools/WebKitTestRunner/GNUmakefile.am	2014-02-04 00:10:28.219478239 +0400
@@ -71,6 +71,7 @@
 endif
 
 webkittestrunner_built_sources += \
+	DerivedSources/InjectedBundle/$(am__dirstamp) \
 	DerivedSources/InjectedBundle/JSAccessibilityController.cpp \
 	DerivedSources/InjectedBundle/JSAccessibilityTextMarker.cpp \
 	DerivedSources/InjectedBundle/JSAccessibilityTextMarkerRange.cpp \
